tests/console/migration_via_opensuse_migration_tool.pm

```diff
@@ -0,0 +1,38 @@
+# SUSE's openQA tests
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: opensuse-migration-tool
+# Summary: A command-line tool to simplify upgrades and migrations across openSUSE distributions
+#              — including Leap, Tumbleweed, Slowroll, and even migrations from Leap to SLE.
+#              - https://github.com/openSUSE/opensuse-migration-tool
+#
+# Maintainer: QE Core <qe-core@suse.de>
+
+use base "opensusebasetest";
+use testapi;
+use utils qw(quit_packagekit zypper_call);
+use power_action_utils 'power_action';
+
+sub run {
+    select_console 'root-console';
+    quit_packagekit;
+    zypper_call('in opensuse-migration-tool');
+
+    enter_cmd "opensuse-migration-tool | tee /dev/$serialdev";
+    assert_screen 'select_the_migration_target';
+    my $migration_target_version = "target_version" . "-" . get_required_var('DISTRI') . "_" . get_required_var('VERSION');
+    send_key_until_needlematch "$migration_target_version", 'down', 5, 2;
+    send_key 'ret';
+    send_key 'ret' if (check_screen 'disable_3rd_party_repositories', 10);
```

Comment:
Please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details